src/location.rs

```diff
@@ -59,27 +32,163 @@ where
         return Err(err_msg("Could not grab pointer"));
     }
 
+    Ok(())
+}
+
+// Updates the cursor for an _already grabbed pointer_
+fn update_cursor(conn: &Connection, cursor: u32) -> Result<(), Error> {
+    xproto::change_active_pointer_grab_checked(conn, cursor, xbase::CURRENT_TIME, GRAB_MASK)
+        .request_check()?;
+
+    Ok(())
+}
+
+// Creates a new `XcursorImage`, draws the picker into it and loads it, returning the id for a `Cursor`
+fn create_new_cursor(
```

Comment:
So `xcolor` no longer uses an X window to draw the preview. Instead, the preview _is_ the mouse cursor!

This may seem like a strange way to do it, but it is a workaround for the fact that `XGetImage` will return empty data for drawable (or window) that's obscured by another window (or offscreen). What this means for `xcolor` is that the picker could not be a window and at the same time request pixels _underneath_ it.

While it's a great workaround for that, it's also _a lot_ more performant than the previous implementation (I presume this is because X doesn't need to reposition a window, and repositioning the mouse is already a highly optimised operation).